testdata/testoutput2

```diff
@@ -13476,6 +13476,29 @@ No match
     Alfred Winifred
 No match
 
+/[[:<:]]+red/B
```

Comment:
I haven't changed that behaviour. It's existing behaviour in PCRE2, and I was surprised about it too - I saw it because `[[:<:]]` is handled by the code which parses character classes. I couldn't quite believe it, so I wrote a test to be sure, and then saw that there wasn't any existing test. So I'm committing the test.

If we want to change the behaviour, we'd better do it in a separate PR, since it's unrelated.

Reply:
Extra experimentation: you cannot have `^+` or any quantifier following the quantifiers which use special metacharacters.

But for legacy reasons (and it's documented by Philip) it seems you can have `(?=a)+`, in which case the quantifier is (entirely? nearly?) a no-effect modifier.

Reply:
The story was the opposite. Quantifying the assertion was not possible, then somebody come and wanted the `?` support. Doing them was a lot of work in jit :(